source/common/stats/stats_impl.cc

```diff
@@ -353,20 +335,31 @@ void HeapRawStatDataAllocator::free(RawStatData& data) {
   ::free(&data);
 }
 
-void RawStatData::initialize(absl::string_view key) {
+void RawStatData::initialize(absl::string_view key, uint64_t xfer_size) {
   ASSERT(!initialized());
-  if (key.size() > Stats::RawStatData::maxNameLength()) {
+  ref_count_ = 1;
+  memcpy(name_, key.data(), xfer_size);
+  name_[xfer_size] = '\0';
+}
+
+void RawStatData::checkAndInit(absl::string_view key, uint64_t num_bytes_allocated) {
+  uint64_t xfer_size = key.size();
+  ASSERT(sizeof(RawStatData) + xfer_size + 1 <= num_bytes_allocated);
```

Comment:
It seems like we compute the in-memory size of a particular RawStatData often.  Do you think it's worth making some utility function that takes the length of the string (without the null terminator) and computes the size so we make sure we do it the same way every time?

Reply:
I like this note. I did some refactoring work in https://github.com/envoyproxy/envoy/pull/3629/commits/aeb9251d33b5326df1dbe7348a600012a3e959a2.